kolibri/core/auth/models.py

```diff
@@ -634,6 +636,43 @@ def create_superuser(self, username, password, facility=None, full_name=None):
         )
         return superuser
 
+    def get_or_create_os_user(self, facility=None):
+        """
+        Returns a FacilityUser object for the current OS user.
+        If the user does not exist in the database, it is created.
+        """
+        try:
+            os_username = interface.get_username()
+        except NotImplementedError:
+            return None
+        if not os_username:
+            return None
+        from kolibri.core.device.models import OSUser
+
+        try:
+            os_user = OSUser.objects.get(os_username=os_username)
+            return os_user.user
+        except OSUser.DoesNotExist:
+            password = uuid4().hex
+            user = None
+            for i in range(0, 10):
+                try:
+                    with transaction.atomic():
+                        user = self.create_user(
+                            "{}{}".format("_" * i, os_username),
+                            password=password,
```

Comment:
Instead of generating a random password, we could do `user.set_unusable_password()`. From there, the only way to authenticate (by default) is through the app's desktop user interface. Kolibri handles users with unusable passwords quite nicely. For instance, if you are logged in a such a user, it allows you to set a password (from the _Change password_ dialog) without asking you for an existing password.

Reply:
True - just trying to think if that will have any unintended consequences anywhere along the line.

Reply:
We also already have the explicit "NOT_SPECIFIED" dummy password (which isn't hashed etc, just stored raw, [from what I see](https://github.com/learningequality/kolibri/blob/9e53513c43d1c08d66911204dd4b1e80e0eeb5d4/kolibri/core/auth/api.py#L693), so it likely behaves equivalently to `set_unusable_password`, which stores something like `"!" + random_digits`, which will also never be the output of a hashing alg)

Reply:
Closing the loop here - I think using `NOT_SPECIFIED` has a distinct advantage that it would prompt a user to set a password if they merged their OS user into a facility, but then they would still be able to do passwordless login within their own app.

The main thing is it means we have to set the quick start facility type to passwordless so that we don't prompt users to set their password.

Reply:
Hrm, now I think about it - this might be a reason not to do this - and we should add an alternative dummy password.

Reply:
And.... changing my mind once again! We only do this prompting if you have logged in via the SessionViewset create method, so this would not be needed. `NOT_SPECIFIED` it is!